### ISSUE TITLE
Advice against "constant handle"?

### ISSUE THREAD

**Issue by Luashine** [lep/jassdoc#85]
As a part of #73 I was going to add this note, but decided to look further into "why":

> +@note You should not initialize game sounds as constants  in Jass ([why?](https://web.archive.org/web/20160803172634/http://wc3jass.com/4513/help-a-trigger-that-fires-when-game-is-restarted/))

In a reply there moyack says that the problem he had was due to a `constant sound` handle. He had an issue after loading a saved game.

Searching further, all I really found were posts on hive [like this](https://www.hiveworkshop.com/threads/displaytexttoplayer-doesnt-work.252833/post-2535650):

> You shouldn't use handle variables as constant globals, make them nonconstant.

And then [the only interesting discussion here](https://www.hiveworkshop.com/threads/if-i-want-to-use-and-learned-vjass-can-i-find-basic-tutorials.216009/post-2148582), where nobody understands the logic of constant vs non-constant functions, and for handles only the issue with "wc3mapoptimizer" is pointed out (wrongly constant-folding handle variables). Though that's different from whatever bug moyack had after game load.

---

Basically, does anybody know what constants are good for (except basic types like int)? Should we add a general warning to `handle` type? (and then each descendant type too?) to never save handles as a constant?
